test/jdk/java/nio/file/spi/testapp/testapp/Main.java

```diff
@@ -58,5 +58,10 @@ public static void main(String[] args) throws Exception {
             throw new RuntimeException("'path' not in default file system");
         if (!path.equals(foo))
             throw new RuntimeException(path + " not equal to " + foo);
+
+        // exercise the file type detector
+        String fileType = Files.probeContentType(Path.of("."));
+        if (fileType != null)
+            throw new RuntimeException("File type non-null: " + fileType);
```

Comment:
Calling Files.probeContentType is enough to exercise the code, I would be worried that it returns something based on the name of the directory, is that possible?

Reply:
Returned value ignored in 1be6091.